packages/backend/src/peripherals/coingecko/CoingeckoQueryService.ts

```diff
@@ -0,0 +1,33 @@
+import { CoingeckoClient, UnixTime } from "@l2beat/common";
+
+export enum Granularity {
+    Hourly,
+    Daily
+}
+
+export class CoingeckoQueryService {
+    constructor(
+       private coingeckoClient: CoingeckoClient
+    ){}
+
+}
+
+export function generateTimestampsList(
+    from: UnixTime,
+    to: UnixTime,
+    granularity: Granularity
```

Comment:
Personally I prefer string literals to enums. There a a few reasons:
1. Less imports. You just write `'hourly'` instead of `import Granularity; Granularity.Hourly`
2. Easier to debug. If you console.log and enum you'll find that `Hourly = 1` and `Daily = 2`. With string literals what you see is what you get.

One small drawback is that because you don't have a separate type (you could but it mostly isn't worth it) you need to write `granularity: 'daily' | 'hourly'` on every function that uses it, but if it turns out too frustrating you could have a `type Granularity = 'daily' | 'hourly'` and use that one. It still wouldn't need to be imported by the user.